<h1 align="center">
ISM 6413 Homework Assignment 1
</h1>

### Problem Description

SOP (Search Optimization Pro) sells online digital advertising to local business via its sales associates. When a sales associate approaches a local business, they try to convince the local business to let SOP use its experience to manage the business' online advertisements on Google, Facebook, etc. If the sales associate is successful, the business signs a contract with SOP, and they agree on the terms. The contract includes following details: 

- Monthly advertising budget (B): This is the maximum amount SOP will spend on behalf of the business each month.
- Digital Advertising Setup Fee, Annual (S): \$3000-5000; , negotiable and varies according to each contract's terms and the business' complexity.
- Monthly Management Fee (F): $\$2500 + 15$\% of Monthly Ad Spend, where \$2500 is the fixed fee and 15\% is the variable fee rate. The equation can be written as:

\begin{equation}
F = 2500 + 0.15\times B
\end{equation}

Based on the terms of the contract we can then estimate the total annual revenue (TAR) earned from a new business as:  

\begin{equation}
TAR = S + 12 \times F
\end{equation}

Once a contract is signed and underway, SOP pays annual commission (C) to its sales associate who delivered the contract. The commission rate is set as flat 5\% of the total annual revenue of the contract. $C = TAR \times 0.05$

You are asked to write a program that SOP can use to calculate F, TAR and C given B and S for each contracted sale. 

# --------------------------------------------------

#### In the following cell, the code is asking to enter your name and surname. You don't need to write any lines within the cell. Just enter your name and surname (after running the cell).

### (5 Points)

In [59]:
name = input("Enter your name and surname here: ")

Enter your name and surname here:  Suraj Gurung


#### Generate 3 constants for the Fixed Fee, Variable Fee Rate, and Comission Rate. Name them fixed_fee, variable_fee_rate, and comission_rate consecutively.

### (10 Points)

In [60]:
months_in_year = 12 # No need to change anything here

# Write your code here (~3 lines)
fixed_fee = 2500
variable_fee_rate = 0.15
comission_rate = 0.05

#### Request an input from the user for the name of the sales associate by writing "Enter the name of the sales associate: "

#### Save this name into a variable called A. 

#### Once you write the code and the input is requested you can enter the name of the Sales Associate as Brian Griffin.

### (10 Points) 

In [61]:
print("Sales and Sales Associate Commissin Tracker by", name) # no need to change anything here
# Write your code here (~ 1 line)
# Request input for the name of the sales associate
A = input("Enter the name of the sales associate: ")

Sales and Sales Associate Commissin Tracker by Suraj Gurung


Enter the name of the sales associate:  Brian Griffin


#### Request an input from the user by writing "What is the name of the business for this contract? "

#### Save this name into a variable called N. 

#### Once you write the code and the input is requested enter the name of the business as Family Dog Training LLC. 

### (5 Points)

In [62]:
# Write your code here (~ 1 line)

#Request input for the name of the business
N = input("What is the name of the business for the contract: ")


What is the name of the business for the contract:  Family Dog Training LLC


#### Request the Monthly Advertising Budget by writing "What is the monthly advertising budget (B)? "

#### Save this information into a variable called B.

#### Once you write the code and the input is requested enter 2300 as the advertising budget.

### (10 Points)

In [63]:
# Write your code here (~1 line)
# Hint: Don't forget to convert the number into a float or integer.

# Request input for Monthly Advertising Budget
B = input("What is the monthly advertising budget: ")

# Convert it into float
B = float(B)

What is the monthly advertising budget:  2300


#### Request the Digital Advertising Setup Fee by writing "What is the agreed Digital Advertising Setup Fee (S)? "

#### Save this information into a variable called S.

#### Once you write the code and the input is requested enter 3450 as the set up fee.

### (10 Points)

In [64]:
# Write your code here (~1 line)
# Hint: Don't forget to convert the number into a float or integer.

# Request input for Digital Advertising Setup Fee
S = input("What is the agree Digital Advertising Setup Fee: ")

# Convert it into float
S = float(S)

What is the agree Digital Advertising Setup Fee:  3450


#### Calculate TAR and C using the equations given above. Use the constants/variables we created (S, f (generated below), months_in_year, and comission_rate) during these calculations. 

### (20 Points)

In [67]:
# No need to change anything in the following line. If you are receiving an error from it, it is probably 
# because you used a different name for generating the corresponding variables,
# or because you did not convert the user input into a float or integer,
# or because you did not run some of the cells above.
f = fixed_fee + variable_fee_rate * B
print ('f = {} Management Fee'.format(f))

# Write your code here (~2 lines)
TAR = S + months_in_year * f
print('TAR = {} Total Annual Revenue'.format(TAR))

C = TAR * comission_rate
print('C = {} Comission'.format(C))

f = 2845.0 Management Fee
TAR = 37590.0 Total Annual Revenue
C = 1879.5 Comission


#### Complete the following code using the constants and variables you created above such that it prints the following:

Given $B = \$2300.00$, $S = \$3450.00$ for Family Dog Training LLC. 

Your monthly management fee will be $\$2845.00$ for a total revenue of $\$37590.00$.

Associate Brian Griffin will earn $\$1879.50$.

### (30 Points)

In [68]:
# Enter the required information within the paranthesis of format()
# Hint: You need to provide 7 entries.
print("""Given B = ${:.2f}, S = ${:.2f} for {}.\n
Your monthly management fee will be ${:.2f} for a total revenue of ${:.2f}.\n
Associate {} will earn ${:.2f}.""".format(B, S, N, f, TAR, A, C ))


Given B = $2300.00, S = $3450.00 for Family Dog Training LLC.

Your monthly management fee will be $2845.00 for a total revenue of $37590.00.

Associate Brian Griffin will earn $1879.50.
